src/hipscat_import/index/map_reduce.py

```diff
@@ -31,16 +31,46 @@ def create_index(args):
     )
 
     if args.include_order_pixel:
-        ## Take out the hive dictionary behavior.
+        ## Take out the hive dictionary behavior that turns these into int32.
         data["Norder"] = data["Norder"].astype(np.uint8)
         data["Dir"] = data["Dir"].astype(np.uint64)
         data["Npix"] = data["Npix"].astype(np.uint64)
+
+    # There are some silly dask things happening here:
+    # - Turn the existing index column into a regular column
+    # - If that had been the _hipscat_index, and we don't want it anymore, drop it
+    # - Create a new index, using our target indexing_column.
+    #   Use division hints if provided.
     data = data.reset_index()
     if not args.include_hipscat_index:
         data = data.drop(columns=[HIPSCAT_ID_COLUMN])
-    data = data.drop_duplicates()
+
+    if args.divisions is not None and len(args.divisions) > 2:
+        data = data.set_index(args.indexing_column, divisions=args.divisions)
+    else:
+        # Try to avoid this! It's expensive! See:
+        # https://docs.dask.org/en/latest/generated/dask.dataframe.DataFrame.set_index.html
+        data = data.set_index(args.indexing_column)
+
+    if args.drop_duplicates:
+        # More dask things:
+        # - Repartition the whole dataset to account for limited memory in
+        #   pyarrow in the drop_duplicates implementation (
+        #   "array cannot contain more than 2147483646 bytes")
+        # - Reset the index, so the indexing_column values can be considered
+        #   when de-duping.
+        # - Drop duplicate rows
+        # - Set the index back to our indexing_column, but this time, the
+        #   values are still sorted so it's cheaper.
+        data = (
+            data.repartition(partition_size=1_000_000_000)
+            .reset_index()
+            .drop_duplicates()
+            .set_index(args.indexing_column, sorted=True)
+        )
+
+    # Now just write it out to leaf parquet files!
     data = data.repartition(partition_size=args.compute_partition_size)
```

Comment:
I wonder if we can get away without doing a repartition after the `set_index`? That the `set_index` method has the `npartitions` and `partition_size` parameters. Maybe doing the repartitioning with those as part of the set index call could be quicker than having to do two passes over the data for repartitioning? It could just be doing a separate repartition call under the hood with those parameters in `set_index` so it might not make a difference.

Reply:
Also is it ok that here if the user specifies `divisions` in the args and it's used to set the index, with this repartition I think it will ignore those divisions and just do the divisions to the partition_size? I don't think that's necessarily a problem, but as a user it might be confusing to set the divisions and then have the output files not follow those divisions.